src/test/java/net/snowflake/client/jdbc/FileUploaderLatestIT.java

```diff
@@ -867,4 +872,45 @@ public void testUploadWithTildeInPath() throws SQLException, IOException {
       FileUtils.deleteDirectory(subDir.toFile());
     }
   }
+
+  /**
+   * Test fix for if a folder has been deleted before calling FileUtils.listFiles we ignore that
+   * directory. Fix available after version 3.16.0.
+   *
+   * @throws Exception
+   */
+  @Test
+  public void testDeleteDirectoryBeforeListFilesWithWildCard() throws Exception {
+    folder.newFile("TestFileA");
+    folder.newFile("TestFileB");
```

Comment:
let's add samples also with some nested directories

Reply:
I got the impression from [this comment](https://github.com/snowflakedb/snowflake-jdbc/blob/b765408004ccf44efa325260ef07b72b49841c28/src/main/java/net/snowflake/client/jdbc/SnowflakeFileTransferAgent.java#L1953) that we don't check nested directories for pattern matches but I will test it out.

Reply:
Based on the docs for [FileUtils.listFiles](https://commons.apache.org/proper/commons-io/javadocs/api-2.5/org/apache/commons/io/FileUtils.html#listFiles(java.io.File,%20org.apache.commons.io.filefilter.IOFileFilter,%20org.apache.commons.io.filefilter.IOFileFilter)) we pass null for the dirFilter parameter so we don't actually go through the subdirectories.